codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -395,14 +386,11 @@ public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorC
             verifyPrefaceReceived();
 
             Http2Stream stream = connection.requireStream(streamId);
-            verifyRstStreamNotReceived(stream);
             if (stream.state() == CLOSED) {
                 // RstStream frames must be ignored for closed streams.
                 return;
             }
 
-            stream.resetReceived();
-
             listener.onRstStreamRead(ctx, streamId, errorCode);
```

Comment:
Do we need to notify the listener here when they can already listen to the stream for closure? Separate issue?